spec/index.bs

```diff
@@ -14461,6 +14493,20 @@ An <dfn dfn>Origin2D</dfn> is a {{GPUOrigin2D}}.
         either {{GPUOrigin2DDict}}.{{GPUOrigin2DDict/y}}
         or the second item of the sequence or 0 if it isn't present.
 </div>
+<div algorithm>
+    <dfn abstract-op>validate GPUOrigin2D shape</dfn>(origin)
+
+    **Arguments:**
+
+    - |origin|: The {{GPUOrigin2D}} to validate.
+
+   **Returns:** {{undefined}}
+
+    1. Throw a {{TypeError}} if:
+
+       - |origin| is a sequence, and
+       - |origin|.length &lt; 0 or |origin|.length &gt; > 2.
```

Comment:
Is it possible to create a sequence with a negative length? (I mean... it's JavaScript so I'm willing to believe in unspeakable language atrocities...)

Assuming that it's not, though, I think we can simplify this to just:
`Throw a {{TypeError}} if |origin| is a sequence and |origin|.length &gt; 2.`